index.js

```diff
@@ -54,32 +52,36 @@ function extract(mbTilesPath, geojson, propName) {
                 zxyStream.pause();
                 paused = true;
             }
+            var tileBBox = [unproject(z, x, y)[0], unproject(z, x+1, y+1)[1],
+                        unproject(z, x + 1, y + 1)[0], unproject(z, x, y)[1]];
+            var result = query.bbox(tileBBox, true);
 
-            var result = query(unproject(z, x + 0.5, y + 0.5));
-
-            if (!result) {
+            if (!result.length) {
                 process.nextTick(tileSaved);
             } else {
-                var extractName = toFileName(result[propName]);
-
-                if (extracts[extractName] && writable[extractName]) {
-                    saveTile(extracts[extractName], z, x, y);
-
-                } else {
-
-                    writeQueue[extractName] = writeQueue[extractName] || [];
-                    writeQueue[extractName].push([z, x, y]);
-
-                    if (!extracts[extractName]) {
-                        writeExtract(extractName, function () {
-                            writable[extractName] = true;
-                            while (writeQueue[extractName].length) {
-                                var t = writeQueue[extractName].pop();
-                                saveTile(extracts[extractName], t[0], t[1], t[2]);
-                            }
-                        });
-                    }
-                }
+                if (result.length > 1) tilesGot += result.length - 1;
```

Comment:
This is to ensure that `tilesGot` is at part with `tilesDone`, since sometimes `tilesDone` will be incremented twice when a tile contains more than one polygon, and is part of more than one resulting mbtiles.